pkg/remote/client_test.go

```diff
@@ -172,6 +175,40 @@ func TestNewWriteRequestBody(t *testing.T) {
 	assert.Contains(t, string(b), `label1`)
 }
 
+func TestNewWriteRequestBodyWithStaleMarker(t *testing.T) {
+	t.Parallel()
+
+	ts := []*prompb.TimeSeries{
+		{
+			Labels: []*prompb.Label{{Name: "label1", Value: "val1"}},
+			Samples: []*prompb.Sample{{
+				Value:     math.Float64frombits(0x7ff0000000000002),
+				Timestamp: time.Now().UnixMilli(),
+			}},
+		},
+	}
+	b, err := newWriteRequestBody(ts)
+	require.NoError(t, err)
+	require.NotEmpty(t, b)
+
+	sb, err := snappy.Decode(nil, b)
+	require.NoError(t, err)
+
+	var series prompb.WriteRequest
+	err = proto.Unmarshal(sb, &series)
+	require.NoError(t, err)
+	require.NotEmpty(t, series.Timeseries[0])
+	require.NotEmpty(t, series.Timeseries[0].Samples)
+
+	isStaleNaN := func(v float64) bool {
+		return math.Float64bits(v) == uint64(0x7ff0000000000002)
+	}
```

Comment:
Same with this. It would be better as an exported function in some common package.